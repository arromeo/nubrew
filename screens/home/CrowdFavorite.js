import React from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default class CrowdFavorite extends React.Component {
  render() {
    return (
      <TouchableOpacity 
        style={[styles.recommendationContainer, styles.homeScreenFilename]}
        onPress={() => {
          this.props.navigate('CrowdFav');
        }}>
        <Ionicons style={styles.buttonIcon} name="md-contacts" size={25} color="#FFBC02"/>
        <Text style={styles.buttonLabel}>  Crowd Favorites</Text>
      </TouchableOpacity>
    )
  }
}
const styles = StyleSheet.create({
  recommendationContainer: {
    flexDirection: "row",
    marginTop: 10,
    marginBottom: 10,
    padding: 5,
    width: '100%',
    flex: 0.2,
    borderRadius: 25,
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#61170E',
  },
  buttonLabel: {
    color: '#FFBC02'
  },
  buttonIcon: {
    paddingLeft: 8
  }
})