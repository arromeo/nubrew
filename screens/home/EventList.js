import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import { MonoText } from '../../components/StyledText';

export default class EventList extends React.Component {
  render() {
    
    return (
      <View style={styles.contentContainer}>
        <FlatList
          data={this.props.data}
          keyExtractor={item => item.id.toString()}
          renderItem={({item}) => 
          <TouchableOpacity
            style={[styles.eventContainer, styles.homeScreenFilename]}
            onPress={() => {
              this.props.navigate('Detail', {
                id: item.id,
                category: "Event"
              });
              this.props.searchDatabase(item.id.toString(), "HighlightEvent");
            }}>
            <View style={[styles.eventDetailsContainer, styles.homeScreenFilename]}>
              <Text>{item.event_name} at {item.store_name}</Text>
              <MonoText style={styles.codeHighlightText}>{item.details.split('').slice(0, 60).join("")}...</MonoText>
            </View>
          </TouchableOpacity>
        }/>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  contentContainer: {
    paddingTop: 15,
    paddingBottom: 15,
    borderWidth: 0.5,
    borderColor: "black",
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "row",
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
  eventDetailsContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
  },
  homeScreenFilename: {
    marginVertical: 7,
  },
});
