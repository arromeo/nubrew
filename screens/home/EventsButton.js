import React from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default class RecommendedButton extends React.Component {
  render() {
    return (
      <TouchableOpacity 
        style={styles.recommendationContainer}
        onPress={() => {
          this.props.navigate('Events');
        }}>
        <Ionicons style={styles.buttonIcon} name="md-notifications" size={25} color="#FFBC02"/>
        <Text style={styles.buttonLabel}>  Local Events</Text>
      </TouchableOpacity>
    )
  }
}
const styles = StyleSheet.create({
  recommendationContainer: {
    flexDirection: "row",
    marginTop: 10,
    marginBottom: 10,
    padding: 5,
    flex: 0.2,
    width: '100%',
    borderRadius: 25,
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#61170E',
  },
  buttonLabel: {
    color: '#FFBC02'
  },
  buttonIcon: {
    paddingLeft: 8
  }
})