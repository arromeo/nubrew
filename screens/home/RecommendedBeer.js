import React from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  Image,
  View
} from 'react-native';
import { MonoText } from '../../components/StyledText';

export default class RecommendedBeer extends React.Component {
  render() {
    const recommendedBeer = this.props.data;
    return (
      <View style={{flex: 1, justifyContent: 'flex-start'}}>
        <Text style={styles.contentHeader}>Featured Beer</Text>
        <TouchableOpacity 
          style={[styles.recommendationContainer, styles.homeScreenFilename]}
          onPress={() => {
            this.props.navigate('Detail', {
              id: recommendedBeer.beer_id,
              category: "Beer",
            });
          }}>
                  <Image
                    style={{width: 100, height: 100}}
                    source={{uri: recommendedBeer.img_url }}
                    />
          <Text style={{fontWeight: "bold"}}>{recommendedBeer.beer_name}</Text>
          <Text style={styles.beerDetails}>{recommendedBeer.brewery_name}</Text>
          <Text style={styles.beerDetails}>IBU: {recommendedBeer.ibu}</Text>
          <Text style={styles.beerDetails}>ABV: {recommendedBeer.abv}</Text>
        </TouchableOpacity>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  recommendationContainer: {
    flex: 1,
    alignItems: 'center'
  },
  contentHeader: {
    textDecorationLine: "underline",
    textAlign: 'center',
    fontWeight: "bold",
    color: "#61170E",
    marginLeft: 10,
    marginTop: 12,
    marginBottom: 10
  },
  beerDetails: {
    fontSize: 12,
    textAlign: "center"
  }
});
