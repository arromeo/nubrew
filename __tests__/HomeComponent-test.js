import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import '../__mock__/xhr-mock';

import CrowdFavorite from '../screens/home/CrowdFavorite';
import EventList from '../screens/home/EventList';
import EventsButton from '../screens/home/EventsButton';
import RecommendedBeer from '../screens/home/RecommendedBeer';
import RecommendedButton from '../screens/home/RecommendedButton';

import props from '../__mock__/props';

describe('Home page components render correctly', () => {
  it('renders the CrowdFavorite component', () => {
    const tree = renderer.create(<CrowdFavorite navigate={props.navigation.navigate}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the EventList component', () => {
    const tree = renderer.create(<EventList navigate={props.navigation.navigate} data={props.data}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the EventsButton component', () => {
    const tree = renderer.create(<EventsButton navigate={props.navigation.navigate}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the RecommendedBeer component', () => {
    const tree = renderer.create(<RecommendedBeer data={props.data} navigate={props.navigation.navigate}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders RecommendedButton component', () => {
    const tree = renderer.create(<RecommendedButton navigate={props.navigation.navigate}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})