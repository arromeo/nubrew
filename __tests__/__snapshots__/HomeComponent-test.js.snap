// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page components render correctly renders RecommendedButton component 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#61170E",
      "borderRadius": 25,
      "flex": 0.2,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 10,
      "marginTop": 10,
      "opacity": 1,
      "padding": 5,
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#FFBC02",
      }
    }
  >
      Recommendations
  </Text>
</View>
`;

exports[`Home page components render correctly renders the CrowdFavorite component 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#61170E",
      "borderRadius": 25,
      "flex": 0.2,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 10,
      "marginTop": 10,
      "opacity": 1,
      "padding": 5,
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#FFBC02",
      }
    }
  >
      Crowd Favorites
  </Text>
</View>
`;

exports[`Home page components render correctly renders the EventList component 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingBottom": 5,
    }
  }
/>
`;

exports[`Home page components render correctly renders the EventsButton component 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#61170E",
      "borderRadius": 25,
      "flex": 0.2,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 10,
      "marginTop": 10,
      "opacity": 1,
      "padding": 5,
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#FFBC02",
      }
    }
  >
      Local Events
  </Text>
</View>
`;

exports[`Home page components render correctly renders the RecommendedBeer component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#61170E",
        "fontWeight": "bold",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginTop": 12,
        "textAlign": "center",
        "textDecorationLine": "underline",
      }
    }
  >
    Featured Beer
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      IBU: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      ABV: 
    </Text>
  </View>
</View>
`;
